install/gcp/installer/util/gcloud.py

```diff
@@ -45,11 +47,47 @@ def get_gcloud_info():
             props = config.get('properties', {})
             metrics = props.get('metrics', {})
             is_devshell = metrics.get('environment') == 'devshell'
+            is_service_account = 'iam.gserviceaccount.com' in authed_user
             print('Read gcloud info: Success')
         except ValueError as verr:
             print(verr)
             sys.exit(1)
-    return project_id, authed_user, is_devshell
+    return project_id, authed_user, is_devshell, is_service_account
+
+def _get_service_account_json_path():
+    """Search in the environment variables for Google Credentials
+    Returns:
+        str: The value of the first non-empty environment variable
+    """
+
+    if 'GOOGLE_CREDENTIALS' in os.environ:
```

Comment:
I am wondering if it would be better to use a FLAG to tell the installer where the key file is, instead of having this method with these logic here?  Such as `--service-account-key-path`.  It would simplify things on our side, with less logic.  Would a flag like this work on your side?

Then above, you can check if this flag is configured, then authed_user is service account.

```
if not service_account_key_path:
    self.check_if_authed_user_in_domain(
        self.organization_id, authed_user)
else:
    gcloud.activate_service_account(authed_user)
```

How do you think?